kafka-connect-s3/src/main/java/io/confluent/connect/s3/S3SinkConnectorConfig.java

```diff
@@ -427,8 +433,23 @@ public int getPartSize() {
   @SuppressWarnings("unchecked")
   public AWSCredentialsProvider getCredentialsProvider() {
     try {
-      return ((Class<? extends AWSCredentialsProvider>)
-                  getClass(S3SinkConnectorConfig.CREDENTIALS_PROVIDER_CLASS_CONFIG)).newInstance();
+      AWSCredentialsProvider provider = ((Class<? extends AWSCredentialsProvider>)
+          getClass(S3SinkConnectorConfig.CREDENTIALS_PROVIDER_CLASS_CONFIG)).newInstance();
+
+      if (provider instanceof Configurable) {
+        Map<String, Object> configs = originalsWithPrefix(CREDENTIALS_PROVIDER_CONFIG_PREFIX,
+            false);
+        if (configs.size() == 0) {
+          log.warn(
+              "{} is a configurable credentials provider, "
+                  + "but no configs with prefix {} has been provided",
+              provider.getClass().getSimpleName(),
+              CREDENTIALS_PROVIDER_CONFIG_PREFIX);
+        }
```

Comment:
I think if a credential provider doesn't require properties - it shouldn't implement `Configurable` interface, `AWSCredentialsProvider` will be enough. But if they implemented `Configurable` interface I feel it will be more intuitive and less error prone to remind users to pass properties to it.

Reply:
Another idea - users' implementation might have default values, and in some cases, they still want to override default configs - in this case, we don't need a warning if no configs passed.